test/framework/clusterctl/e2e_config.go

```diff
@@ -245,6 +248,83 @@ type Files struct {
 	TargetName string `json:"targetName,omitempty"`
 }
 
+// ResolveReleases converts release markers to release version.
+func (c *E2EConfig) ResolveReleases() error {
+	scheme, host, err := goproxy.GetSchemeAndHost(os.Getenv("GOPROXY"))
+	if err != nil {
+		return err
+	}
+	if scheme == "" || host == "" {
+		return fmt.Errorf("releasemarker does not support disabling the go proxy: GOPROXY=%q", os.Getenv("GOPROXY"))
+	}
+	goproxyClient := goproxy.NewClient(scheme, host)
+	for i := range c.Providers {
+		provider := &c.Providers[i]
+		for j := range provider.Versions {
+			version := &provider.Versions[j]
+			if version.Type == "url" {
+				releaseMarker := strings.TrimLeft(strings.TrimRight(version.Name, "}"), "{")
+				if !strings.EqualFold(version.Name, releaseMarker) {
+					ver, err := resolveReleaseMarker(releaseMarker, goproxyClient)
+					if err != nil {
+						return fmt.Errorf("failed resolving release path %s with error: %v", version.Name, err)
+					}
+					ver = "v" + ver
+					version.Value = strings.Replace(version.Value, version.Name, ver, 1)
+					version.Name = ver
+				}
+			}
+		}
+	}
+	return nil
+}
+
+// resolveReleaseMarker resolves releaseMarker string to verion string e.g.
+// - Resolves "go://sigs.k8s.io/cluster-api@v1.0" to the latest stable patch release of v1.0.
+// - Resolves "go://sigs.k8s.io/cluster-api@latest-v1.0" to the latest patch release of v.1.0 including rc and pre releases.
+func resolveReleaseMarker(releaseMarker string, goproxyClient *goproxy.Client) (string, error) {
+	parts := strings.Split(releaseMarker, ":")
+	if len(parts) <= 1 {
+		return "", errors.Errorf("Invalid release url")
+	}
+
+	gomoduleParts := strings.Split(parts[1], "@")
+	if len(gomoduleParts) < 2 {
+		return "", errors.Errorf("Invalid release url, go module or version missing")
+	}
+	gomodule := gomoduleParts[0]
+	version := strings.ReplaceAll(gomoduleParts[1], "v", "")
+	includePrereleases := false
+	if strings.HasPrefix(version, "latest-") {
+		includePrereleases = true
+	}
+	version = strings.TrimPrefix(version, "latest-")
+
+	minSemVersion, err := semver.Parse(version + ".0")
+	if err != nil {
+		return "", err
```

Comment:
Let's use proper error wrapping (also at the other places)
```suggestion
		return "", errors.Wrapf(err, "Parsing semver for %s", version+".0")
```

(Note: maybe store `version + ".0"` in a string first)